### PR TITLE
Mob 3465 make project build with latest android studio

### PR DESCRIPTION
**Pull request by cosminBadulescu**
## JIRA
[MOB-3465](https://karhoo.atlassian.net/browse/MOB-3465)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -11,16 +11,14 @@ buildscript {
...
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.6.4'
+        classpath 'com.android.tools.build:gradle:4.2.2'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
-        classpath 'io.fabric.tools:gradle:1.30.0'
         classpath 'com.jaredsburrows:gradle-license-plugin:0.8.1'
         classpath 'com.novoda:bintray-release:0.9.2'
-        classpath 'org.jacoco:org.jacoco.core:0.8.2'
+        classpath 'org.jacoco:org.jacoco.core:0.8.4'
```
Comment (mohadian) [inline build.gradle:21]: version 0.8.7?

File: uisdk/src/test/java/com/karhoo/uisdk/screen/booking/quotes/mocks/QuotesListViewMock.kt [hunk 2]
```diff
@@ -10,44 +10,33 @@ import com.karhoo.uisdk.screen.booking.quotes.QuotesListMVP
...
         calledShowNowResults = show
     }
 
-    override fun hideNoAvailability() {
-    }
+    override fun hideNoAvailability() { /** do nothing **/ }
 
-    override fun showSnackbarError(snackbarConfig: SnackbarConfig) {
-    }
-}
```
Comment (mateid-kr) [inline uisdk/src/test/java/com/karhoo/uisdk/screen/booking/quotes/mocks/QuotesListViewMock.kt:53]: Why the changes on these? For passing the detekt checks?

**Review by mateid-kr — DISMISSED** [review]

File: build.gradle (hunk 1 above)
Comment (cosminBadulescu) [reply build.gradle]: done

File: uisdk/src/test/java/com/karhoo/uisdk/screen/booking/quotes/mocks/QuotesListViewMock.kt (hunk 2 above)
Comment (cosminBadulescu) [reply uisdk/src/test/java/com/karhoo/uisdk/screen/booking/quotes/mocks/QuotesListViewMock.kt:53]: yes, detekt was failing because of empty override functions

**Review by mateid-kr — APPROVED** [review]
